lib/WP_Auth0_LoginManager.php

```diff
@@ -543,16 +543,41 @@ protected function query_vars( $key ) {
   /**
    * Get and store state returned from Auth0
    *
-   * @return object|null
+   * @param bool $decoded - `true` to return decoded state
+   *
+   * @return string|object|null
    */
-  protected function get_state() {
+  protected function get_state( $decoded = FALSE ) {
```

Comment:
why would I need the "encoded" state? URL encoding a state shouldn't be responsibility of this class, but the class who constructs the requests.

Reply:
References base64 encoded, not URL. URL decoding has to happen, this is just basically "do you want the raw value or the object"

Reply:
Why would I need it to be base64 encoded? The form of the `nonce` or `state` values should be obscure. I don't care how you generate it internally as long as it's "safe and secure". (guarantying that it's too hard to repeat itself in a short period of time, etc). 

And TBH, the server doesn't care either. It's a plain text string that it needs to put somewhere so the client can later obtain it and validate it with the one it sent before. So unless you have a reason **why** to have a method supporting both encoded/decoded scenarios, I think it's useless.

Reply:
It's base64 encoded when it's created in `WP_Auth0_Lock10_Options->get_state_obj()` ... more specifically, it's a JSON encoded object that's then base64 encoded. That's what I store in a cookie then validate during the login process, still base64 encoded. Once someone is logged in, then I need that object I created before to get a redirect URL. 

If you look for where `get_state` is used in this class, you'll see what I mean